packages/astro/src/actions/utils.ts

```diff
@@ -25,3 +27,46 @@ export function createCallAction(context: ActionAPIContext): APIContext['callAct
 		return action(input) as any;
 	};
 }
+
+let didInitLexer = false;
+
+export async function usesActions(fs: typeof fsMod, srcDir: URL) {
+	if (!didInitLexer) await eslexer.init;
+
+	const actionsFile = search(fs, srcDir);
+	if (!actionsFile) return false;
+
+	let contents: string;
+	try {
+		contents = fs.readFileSync(actionsFile, 'utf-8');
+	} catch {
+		return false;
+	}
+
+	const [, exports] = eslexer.parse(contents, actionsFile.pathname);
+	for (const exp of exports) {
+		if (exp.n === 'server') {
+			return true;
+		}
+	}
+	return false;
+}
+
+function search(fs: typeof fsMod, srcDir: URL) {
+	const paths = [
+		'actions.mjs',
+		'actions.js',
+		'actions.mts',
+		'actions.ts',
+		'actions/index.mjs',
+		'actions/index.js',
+		'actions/index.mts',
+		'actions/index.ts',
+	].map((p) => new URL(p, srcDir));
```

Comment:
Why do we use this pattern to look for an action file? This pattern is really similar to `middleware.js` - where users can have `middleware.js` or `middleware/index.js`, and more -  and in that case we use the `moduleLoader` to retrieve the file.

It's not a blocker, but I think we should agree on **one way** to do things internally. 

Using the module loader from vite would allow us also to remove the use of `es-module-lexer`

Reply:
Discussed on discord. We discovered that we can't rely on Vite to check whether actions are used, since we need that information _before_ vite starts to inject routes. We're tabling this for now to find a better solution.